k3k-kubelet/provider/provider.go

```diff
@@ -665,3 +670,37 @@ func getSecretsAndConfigmaps(pod *corev1.Pod) ([]string, []string) {
 	}
 	return secrets, configMaps
 }
+
+// fetchFieldPathAnnotations will retrieve all annotations created by the pod mutator webhook
+// to assign env fieldpaths to pods
+func (p *Provider) fetchFieldPathAnnotations(pod, tPod *v1.Pod) error {
+	for name, value := range pod.Annotations {
+		var (
+			containerIndex int
+			envName        string
+			err            error
+		)
+		if strings.Contains(name, webhook.FieldpathField) {
+			s := strings.SplitN(name, "_", 3)
+			if len(s) != 3 {
+				continue
+			}
+			containerIndex, err = strconv.Atoi(s[1])
+			if err != nil {
+				return err
+			}
+			envName = s[2]
+
+			// re-adding these envs to the pod
```

Comment:
After readding them, should we cleanup the annotations as well? Do we need them anymore?

Reply:
I will remove them from the pods on the host cluster, removing them after they are created on the virtual cluster will require a new controller which I think its an overkill